pkg/util/distance/calc_distance.go

```diff
@@ -130,15 +131,27 @@ func CalcFloatDistance(dim int64, left, right []float32, metric string) ([]float
 
 	distArray := make([]float32, leftNum*rightNum)
 
-	// Multi-threads to calculate distance. TODO: avoid too many go routines
+	// Multi-threads to calculate distance.
 	var waitGroup sync.WaitGroup
-	CalcWorker := func(index int64) {
-		CalcFFBatch(dim, left, index, right, metricUpper, &distArray)
+	calcWorker := func(index interface{}) {
+		i := index.(int64)
+		CalcFFBatch(dim, left, i, right, metricUpper, &distArray)
 		waitGroup.Done()
 	}
+	// avoid too many goroutines by ants pool
+	poolSize := int(leftNum / 3)
+	pool, err := ants.NewPoolWithFunc(poolSize, calcWorker)
```

Comment:
Refer to milvus pkg/util/conc/pool.go, where wraps the ants.pool, you could use the pool by referring to other usages.
Also make this pool as static, not init it every time when we compute distances.

Reply:
ok, i got it

Reply:
hey, @chasingegg last week I was used `sync.Once` to make the pool init once. just like that:
```go
var (
	calcPool         *conc.Pool[any]
	calcPoolInitOnce = new(sync.Once)
)

func initCalcPool() {
	calcPool = conc.NewDefaultPool[any]()
}

func GetCalcPool() *conc.Pool[any] {
	calcPoolInitOnce.Do(initCalcPool)
	return calcPool
}
```
but I find it will going deep into a dead cycle when I run the unit test.

I haven't found the reason for the dead cycle, so I have to make the new pool first in every time.

Reply:
Can I ask how can I reproduce your dead cycle problem?

Reply:
> Can I ask how can I reproduce your dead cycle problem? which unit test?

You can reproduce it by following these steps：
1. copy the `sync.Once` code to make the pool init once.

<img width="1981" alt="image" src="https://github.com/milvus-io/milvus/assets/57799587/04e49298-bbbe-4785-8d2e-000fee73fd82">


2. replace ` pool := conc.NewDefaultPool[any]()` to `pool := GetCalcPool()`

<img width="1099" alt="image" src="https://github.com/milvus-io/milvus/assets/57799587/d709b2ab-f1f9-4450-9bd9-4762ba5466d7">


3. run the ut where in `pkg/util/distance/calc_distance_test.go:155` , and find will go to the dead cycle.

<img width="1199" alt="image" src="https://github.com/milvus-io/milvus/assets/57799587/abf300c9-d63e-4066-8f15-f2e654930f0a">

4. I'm guessing this would be a problem caused by me, so I'm still debuging this problem. But I've been so busy these past few days that I've procrastinated a lot. I will find out the reason why this weekend.

Reply:
Thanks! I will also look into it and hopefully could reproduce the problem